### DISCUSSION TITLE
How to store Git hash in saved config for reproducibility?

### DISCUSSION THREAD

**Discussion by vedal** [Lightning-AI/pytorch-lightning#16281] [General]
Hi!

Is there an elegant way to store the Git hash in the stored hparams stored by self.save_hyperparameters()?
I'm trying to store git hash in stored config when working with LightningCLI, but unsure how to do this in a succinct way that doesn't include environment variables.

**Comment by mauvilsa** [comment]
How about including the git hash in the header comment of the config saved by LightningCLI, similar to how currently includes the version of pytorch-lightning. This could be done as:

```python
class MyCLI(LightningCLI):
    def add_arguments_to_parser(self, parser):
        parser.dump_header.append(f"my_code_version={code_version}")
```

Note that the version of the code is not a configurable hyperparameter, which is why I suggest to include it in a comment.

**Reply by vedal** [reply]
Thank you so much @mauvilsa, this is exactly what I was looking for! :)
I very much appreciate your responsiveness!
